src/snapred/backend/recipe/algorithm/GroupDiffractionCalibration.py

```diff
@@ -272,14 +294,15 @@ def PyExec(self) -> None:
                     InputWorkspace=self.DIFCfinal,
                     OutputWorkspace=self.DIFCprev,
                 )
+            self.mantidSnapper.executeQueue()
+            self.verifyChiSq(diagnosticWSgroup)
```

Comment:
This strikes me as inelegant, but the mantid algos that should be called here don't work with uneven spectra.

Can you pass the index as an argument, so that the warning message produces says which spectrum is failing by its index?

Reply:
I think the preferred method would be to append the tables together in a workspace, then tell users to check the workspace in the case of an error so they can see which peaks to adjust.

Reply:
Or include all of that information (spectrum, peak location, chisq value) for all of the bad peaks inside the message box.